File: server/module/request.js
const request = require('request');
const handler = require('./handler');
const querystring = require('querystring');
const initSearchUrl = (url, param) => (handler.emptyObject(param) ? url : url + '?' + querystring.stringify(param));
exports.get = (url, param, token = {}, isJson) => {
    return new Promise((resolve, reject) => {
        request.get(initSearchUrl(url, param), { headers: token }, (error, response, body) => {
            !error && response.statusCode === 200 ? resolve(isJson ? JSON.parse(body) : body) : reject(error);
        });
    });
};
exports.post = (url, param, token = {}, isJson) => {
    return new Promise((resolve, reject) => {
        request.post(url, { headers: token, json: isJson, body: param }, (error, response, body) => {
            !error && response.statusCode === 200 ? resolve(body) : reject(error);
        });
    });
};
exports.put = (url, param, token = {}, isJson) => {
    return new Promise((resolve, reject) => {
        request.put(url, { headers: token, json: isJson, body: param }, (error, response, body) => {
            !error && response.statusCode === 200 ? resolve(body) : reject(error);
        });
    });
};
exports.delete = (url, param, token = {}, isJson) => {
    return new Promise((resolve, reject) => {
        request.del(url, { headers: token, json: isJson, body: param }, (error, response, body) => {
            !error && response.statusCode === 200 ? resolve(body) : reject(error);
        });
    });
};

File: server-unified/api/module/request.py
import requests
from api.module import handler


def get(url, params=None, token={}, mock=None, timeout=None):
    try:
        response = requests.get(
            url, params=params, headers=token, timeout=timeout)
        if response.status_code == requests.codes.ok:
            return handler.unicode2str(response.json())
        else:
            response.raise_for_status()
    except Exception as error:
        if mock is not None:
            return mock
        else:
            raise requests.ConnectionError(error)


def post(url, params=None, token={}, timeout=None):
    try:
        response = requests.post(
            url, json=params, headers=token, timeout=timeout)
        if response.status_code == requests.codes.ok:
            return handler.unicode2str(response.json())
        else:
            response.raise_for_status()
    except Exception as error:
        raise requests.ConnectionError(error)
